bert.m

```diff
@@ -5,28 +5,75 @@
 %
 %   mdl = bert('Model', modelName) loads the BERT model specified by
 %   modelName.  Supported values for modelName are "base" (default),
-%   "multilingual-cased","medium","small","mini", and "tiny".
+%   "multilingual-cased","medium","small","mini", "tiny", "japanese-base", 
+%   and "japanese-base-wwm"
 
-% Copyright 2021 The MathWorks, Inc.
+% Copyright 2021-2023 The MathWorks, Inc.
 arguments
     nvp.Model (1,1) string {mustBeMember(nvp.Model,[
         "base"
         "multilingual-cased"
         "medium"
         "small"
-        "mini"        
-        "tiny"])} = "base"
+        "mini"
+        "tiny"
+        "japanese-base"
+        "japanese-base-wwm"])} = "base"
+end
+
+switch nvp.Model
+    case "japanese-base"
+        zipFilePath = bert.internal.getSupportFilePath("japanese-base", "bert-base-japanese.zip");
+        modelDir = fullfile(fileparts(zipFilePath), "bert-base-japanese");      
+        unzip(zipFilePath, modelDir);
+        % Build the tokenizer
+        btok = bert.tokenizer.internal.TokenizedDocumentTokenizer("Language","ja","TokenizeMethod","mecab",IgnoreCase=false);
+        vocabFile = fullfile(modelDir, "vocab.txt");
+        ftok = bert.tokenizer.internal.FullTokenizer(vocabFile,BasicTokenizer=btok);
+        tok = bert.tokenizer.BERTTokenizer(vocabFile,FullTokenizer=ftok);
+        % Build the model
+        params.Weights = load(fullfile(modelDir, "weights.mat"));
+        params.Weights = dlupdate(@dlarray,params.Weights);
+        params.Hyperparameters = struct(...
+            NumHeads=12,...
+            NumLayers=12,...
+            NumContext=512,...
+            HiddenSize=768);
+        mdl = struct(...
+            Tokenizer=tok,...
+            Parameters=params);
+    case "japanese-base-wwm"
+        zipFilePath = bert.internal.getSupportFilePath("japanese-base", "bert-base-japanese-whole-word-masking.zip");
+        modelDir = fullfile(fileparts(zipFilePath), "bert-base-japanese-whole-word-masking");      
+        unzip(zipFilePath, modelDir);
+        % Build the tokenizer
+        btok = bert.tokenizer.internal.TokenizedDocumentTokenizer("Language","ja","TokenizeMethod","mecab",IgnoreCase=false);
+        vocabFile = fullfile(modelDir, "vocab.txt");
+        ftok = bert.tokenizer.internal.FullTokenizer(vocabFile,BasicTokenizer=btok);
+        tok = bert.tokenizer.BERTTokenizer(vocabFile,FullTokenizer=ftok);
+        % Build the model
+        params.Weights = load(fullfile(modelDir, "weights.mat"));
+        params.Weights = dlupdate(@dlarray,params.Weights);
+        params.Hyperparameters = struct(...
+            NumHeads=12,...
+            NumLayers=12,...
+            NumContext=512,...
+            HiddenSize=768);
+        mdl = struct(...
+            Tokenizer=tok,...
+            Parameters=params);
```

Comment:
There's a lot of duplicate config to `japanese-base`, which makes sense since they're both BERT-Base models. I think it's fine to make this a helper function that just takes the specific file names you need - e.g. maybe we need to add some extra meta-data to the `Hyperparameters` some day, and then we only have to do that in one place rather than 2.

Reply:
How now I have left it specific to Japanese BERT, but we can make it more general once we add more models.